### PR TITLE
fix: store seeds on their stage's plan, exact free rotations

### PR DESCRIPTION
**Pull request by vboussot**
Stacked on #186 (base `perf/stream-known-statistic`). Two of the items #186 left for later, plus the one that depended on them.

## What changes

1. **fix(patching): seed a stage from the store on its own plan alone.** The store's statistics a streamed stage was seeded with (`Statistics`, `Clip('min', 'max')`, a head `Standardize`) were written into the planning scope, which became the copy's case attribute, the backup every whole-volume load starts from and the header a Save sweep writes. A patch transform then read the case's Mean over its own patch's (0.22 off), a whole load after a plan no longer landed where one that ran no plan did, and a `Standardize` behind a sweep-written Save took the source's Mean (0.31 off). A store seed now rides the stage's plan like a measured one: pushed right before the stage runs, taken back after it unless the whole-volume route leaves the key there. Two declarations on `PatchLocality` say which: `records` (the keys the whole-volume call leaves under those names; `Statistics` none, a `Clip` bound only when saved) and `takes_present` (`Normalize`, `Standardize` take a statistic already in the scope, so behind a stage that records it they need no seed). `[Clip(save_clip_*), Normalize]` and `[Clip, Standardize, Clip, Standardize]` stream again, and the refusal that kept a measured stage whole beside a store seed of the same name is gone.
2. **fix(augmentation): sample a free rotation or scale exactly on a region.** `grid_sample` normalises coordinates by the extent it is handed: a region is handed a window, the whole volume the volume, and the two rounded apart by up to 1.5e-4. The draw now walks the corners itself on the full grid's own indices, so a region lands where the whole volume does, bit for bit; a label map takes the nearest voxel the same way. An axis the map leaves on the grid (the slice axis of an in-plane turn) has one tap, so an in-plane turn walks four corners.

## Measured

- Fuzz of 1 200 random chains (transforms, draws with random probabilities, patch transforms, redraws, random operation orders), every read compared with a manager that never planned: 54 violating chains at #186's head, 32 after commit 1 (all free `Rotate`/`Scale`), 0 after commit 2.
- The corner walk against `grid_sample`, single thread: a 5x64x64 region of a 5x512x512 case 1.5 to 1.9 ms; a whole 32x256x256 volume 191 to 208 ms; a whole 100x512x512 volume 3.4 to 5.4 s. An oblique turn walks eight corners and costs about twice that.
- The REGRID draw's working multiple says the ten volumes its coordinate build peaks at (measured on a 100x512x512 case; it said four, and the old route peaked at twelve).

## Verification

`pixi run test`, `pixi run --environment dev typecheck` and ruff pass on the head. The new tests fail on #186's head.

## Contract

`AGENTS.md` names the two declarations. A stage that leaves its statistic under other names, or not at all, declares `records`; one that takes a present statistic declares `takes_present`. A stage declaring neither keeps the old behaviour: its seed stays, as its whole-volume call would have left it.

## Summary by CodeRabbit

- **New Features**
  - Added an option to constrain 3D rotations to the slice plane.
  - Added visibility into landed channels and streamed-read amplification.
  - Added support for warming streamed statistics before processing.

- **Bug Fixes**
  - Improved spatial resampling accuracy for integer labels and floating-point data, including boundary handling and precision preservation.
  - Fixed streamed rotation and scaling to match whole-volume results.
  - Corrected transformed grid spacing after reloads and resets.
  - Prevented temporary statistics from leaking into later stages or saved metadata.

- **Reliability**
  - Improved reuse and propagation of measured statistics during streamed processing.

### CODE REVIEW

File: konfai/data/augmentation/spatial.py [hunk 1]
```diff
@@ -161,24 +163,72 @@ def _sample_region(
...
+            fraction = fractions[..., axis]
+            lower = window_index(bases[..., axis], full[axis], starts[axis], windows[axis])
+            axis_taps = [(lower, 1.0 - fraction)]
+            if bool(fraction.any()):
+                axis_taps.append(
+                    (window_index(bases[..., axis] + 1, full[axis], starts[axis], windows[axis]), fraction)
+                )
+            taps.append(axis_taps)
+        work = flat if flat.dtype == torch.float32 else flat.to(torch.float32)
+        out = torch.zeros((channels, *shape), dtype=torch.float32, device=flat.device)
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/augmentation/spatial.py]: Fixed in 045e95f: a measuring stage is seeded with every key it declares over a header's, a stage taking one key and measuring another is served from the scope for the first and measured for the second, and a float64 block blends in float64.

File: konfai/data/patching/manager.py [hunk 2]
```diff
@@ -418,6 +434,9 @@ def _keep_measurements(
             scope = scopes[a]
             if all(scope._count_key(key) > depth for key, depth in before.items()):
                 self._measured_statistics[(a, id(stage))] = {key: scope[key] for key in before}
+            elif wanted[a].takes_present and all(depth > 0 for depth in before.values()):
+                # It took what an earlier stage recorded, which the streamed route records there too.
+                self._scope_served.add((a, id(stage)))
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/patching/manager.py]: Fixed in 045e95f: a measuring stage is seeded with every key it declares over a header's, a stage taking one key and measuring another is served from the scope for the first and measured for the second, and a float64 block blends in float64.

File: konfai/data/patching/manager.py [hunk 3]
```diff
@@ -608,31 +627,34 @@ def warm_stream_statistics(self, copies: Sequence[int] = (0,), apply_augmentatio
...
-        channels: list[int] | None = None,
-    ) -> None:
-        missing_stats = [key for key in required_stats if key not in cache_attribute]
-        if not missing_stats:
-            return
+        loc: PatchLocality,
+    ) -> tuple[tuple[str, str], ...]:
+        """The store's statistics a stage is seeded with, as the scope spells them: those its header
+        does not already hold, which a stage takes as the whole-volume route does."""
+        missing = sorted(key for key in loc.stat_keys if key not in cache_attribute)
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/patching/manager.py]: Fixed in 045e95f: a measuring stage is seeded with every key it declares over a header's, a stage taking one key and measuring another is served from the scope for the first and measured for the second, and a float64 block blends in float64.
